hal/src/nRF52840/gsm0710muxer/channel_stream.h

```diff
@@ -217,10 +217,11 @@ inline int MuxerChannelStream<MuxerT>::waitEvent(unsigned flags, unsigned timeou
         if (f &= flags) {
             break;
         }
-        if (timeout > 0 && HAL_Timer_Get_Milli_Seconds() - t >= timeout) {
+        const auto elapsed = HAL_Timer_Get_Milli_Seconds() - t;
+        if (timeout == 0 || elapsed >= timeout) {
```

Comment:
Was this a bug before?  The logic for this conditional has changed.

Reply:
> This PR also fixes a bug in ChannelStream in waitEvent() timeout handling